server/src/main/java/io/crate/planner/consumer/UpdatePlanner.java

```diff
@@ -134,7 +134,7 @@ private static Plan plan(DocTableRelation docTable,
         WhereClauseOptimizer.DetailedQuery detailedQuery = WhereClauseOptimizer.optimize(
             normalizer, query, tableInfo, plannerCtx.transactionContext(), plannerCtx.nodeContext());
 
-        if (detailedQuery.docKeys().isPresent()) {
+        if (detailedQuery.docKeys().isPresent() && detailedQuery.whereClauseConsistsOfPkSymbolsOnly()) {
```

Comment:
I think in the long run, we may want to support the optimized UpdateById plans but adds some kind of projections like we did for the SELECTs by https://github.com/crate/crate/commit/ab5e0a4a74e4b841661333f6aa1e8f79e6737c5c.

As such I would suggest we detect non-pk columns inside the query (which we would need as inputs for a future projection anyhow) in a 2nd phase instead of adding more logic to the EqualityExtractor.
E.g. re-using same logic as used inside the `Get` operator: https://github.com/crate/crate/blob/master/server/src/main/java/io/crate/planner/operators/Get.java#L173

Reply:
Thank you, please check the new commit.